### DISCUSSION TITLE
ConfD dependency in JNC

### DISCUSSION THREAD

**Discussion by muralivenugopal** [tail-f-systems/JNC#133] [General]
Hi,

I'm new to JNC. 
I don't understand the need for tailf ConfD in the Netconf Client. JNC has a sshj client to interact with netconf device. pyang jnc plugin is used to convert the yang to Java POJOs. JNC jar generated is the client app. Can't we use these directly to connected to a netconf device to communicate? why do we need ConfD.

**Comment by martin-volf** [comment]
It is as you write, you should not need ConfD so as to use JNC. The JNC's examples use ConfD as the NETCONF server; but to build and run a NETCONF client application, ConfD is not needed. Do you observe something different?

**Reply by muralivenugopal** [reply]
No, Thanks for the reply. Im was just wondering why ConfD is required when I have an actual SNMP Enabled Device. Im facing issues in packaging the generated Java files into a Jar and use it other projects. Because of the .schema files. Is there any best practice that you suggest?
